libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
At the point this check is happening though, one of two things would happen the way the code reads:

- `$this->connect()` fails and throws an Exception, so this check wouldn't get triggered
- `$this->connection` is an object/resource/whatever-we-check-since-it-is-inconsistent-across-the-API so we try the query and the error gets handled by the existing handlers later in this method

Now that I'm actually looking at this, I'm actually not sure I can see a condition where this check would be reached and hit a failure.  Unless I'm just not making sense of it all in my head.